build_runner/test/generate/build_test.dart

```diff
@@ -361,6 +363,23 @@ void main() {
             });
       });
 
+      test('disallows reading hidden outputs to create a non-hidden output',
```

Comment:
It would be good to add a test for the behavior of not being able to read hidden assets in other packages as well, or merge that into this test.

Reply:
Done